synapse/lib/stormlib/aha.py

```diff
@@ -531,5 +620,223 @@ async def _methPoolSvcDel(self, svcname):
+                    $indices.add($status.nexs_indx)
+                    $known_count = ($known_count + (1))
+                }
+            }
+            if ($lib.len($indices) = 1) {
+                if ($known_count = $lib.len($group_status)) {
+                    return(true)
+                }
+            }
+        }
+
+        function output_status(vname, group_status, printer) {
+            $lib.print($printer.header())
+            $lib.print($vname)
+            for $status in $group_status {
+                $row = (
+                    $status.name,
+                    $status.role,
+                    $status.online,
+                    $status.ready,
+                    $status.host,
+                    $status.port,
+                    $status.version,
+                    $status.nexs_indx
+                )
+                $lib.print($printer.row($row))
+            }
+        }
+
+        $virtual_services = ({})
+        $member_servers = ({})
+
+        for $svc in $lib.aha.list() {
+            $name = $svc.name
+            $svcinfo = $svc.svcinfo
+            $urlinfo = $svcinfo.urlinfo
+            $hostname = $urlinfo.hostname
+
+            if ($name != $hostname) {
+                $virtual_services.$name = $svc
+            } else {
+                $member_servers.$name = $svc
+            }
+        }
+
+        $mirror_groups = ({})
+        for ($vname, $vsvc) in $virtual_services {
+            $vsvc_info = $vsvc.svcinfo
+            $vsvc_iden = $vsvc_info.iden
+            $vsvc_leader = $vsvc_info.leader
+            $vsvc_hostname = $vsvc_info.urlinfo.hostname
+
+            if (not $vsvc_iden or not $vsvc_hostname or not $vsvc_leader) {
+                continue
+            }
+
+            $primary_member = $member_servers.$vsvc_hostname
+            if (not $primary_member) {
+                continue
+            }
+
+            $members = $lib.list($primary_member)
+            for ($mname, $msvc) in $member_servers {
+                if ($mname != $vsvc_hostname) {
+                    $msvc_info = $msvc.svcinfo
+                    if ($msvc_info.iden = $vsvc_iden and $msvc_info.leader = $vsvc_leader) {
+                        $members.append($msvc)
+                    }
+                }
+            }
+
+            if ($lib.len($members) > 1) {
+                $mirror_groups.$vname = $members
+            }
+        }
+
+        for ($vname, $members) in $mirror_groups {
+            $cell_infos = $get_cell_infos($vname)
+            $group_status = $build_status_list($members, $cell_infos)
+            $lib.print('Service Mirror Groups:')
+            $output_status($vname, $group_status, $printer)
+
+            if $check_sync_status($group_status) {
+                $lib.print('Group Status: In Sync')
+            } else {
+                $lib.print(`Group Status: Out of Sync; Waiting {$timeout} seconds for sync...`)
```

Comment:
Seems like we should just print that they're out of sync and let the user run the command again in a minute. I bet this will hang the storm cli and you won't be able to kill it without killing the process.

Reply:
I can change it; @vEpiphyte any additional input here?

Reply:
The Storm CLI hanging is it's own interaction with prompt_toolkit; not the command implementation. ( see SYN-7932 ). We shouldn't mix the two in this PR.

We could add a `--sync` option that would then try the sync if they are **not** already caught up?

That way the default `aha.svc.mirror` would never block; but `--sync` could block until the mirrors are caught up?

Reply:
Yeah I can change it. Maybe `--wait` would be better? I worry that `--sync` implies it will happen which might not necessarily be true. Thoughts?

Reply:
added `--wait` in the latest commit.